src/renderers/shaders/ShaderChunk/uv_pars_vertex.glsl

```diff
@@ -4,3 +4,61 @@
 	uniform vec4 offsetRepeat;
 
 #endif
+
+//@author pailhead
+//for now the most convenient place to attach vert transformation logic in global scope ( before main() )
+#if defined( INSTANCE_TRANSFORM )
+
+mat3 inverse(mat3 m) {
+  float a00 = m[0][0], a01 = m[0][1], a02 = m[0][2];
+  float a10 = m[1][0], a11 = m[1][1], a12 = m[1][2];
+  float a20 = m[2][0], a21 = m[2][1], a22 = m[2][2];
+
+  float b01 = a22 * a11 - a12 * a21;
+  float b11 = -a22 * a10 + a12 * a20;
+  float b21 = a21 * a10 - a11 * a20;
+
+  float det = a00 * b01 + a01 * b11 + a02 * b21;
+
+  return mat3(b01, (-a22 * a01 + a02 * a21), (a12 * a01 - a02 * a11),
+              b11, (a22 * a00 - a02 * a20), (-a12 * a00 + a02 * a10),
+              b21, (-a21 * a00 + a01 * a20), (a11 * a00 - a01 * a10)) / det;
+}
+
+//for dynamic, avoid computing the matrices on the cpu
+attribute vec3 instancePosition;
+attribute vec4 instanceQuaternion;
+attribute vec3 instanceScale;
+
+#if defined( INSTANCE_COLOR )
+  attribute vec3 instanceColor;
+  varying vec3 vInstanceColor;
+#endif
+
+mat4 getInstanceMatrix(){
+
+  vec4 q = instanceQuaternion;
+  vec3 s = instanceScale;
+  vec3 v = instancePosition;
+
+  vec3 q2 = q.xyz + q.xyz;
```

Comment:
per
https://stackoverflow.com/questions/44037202/glsl-division-vs-multiplication

grouped some operations so they can be simd